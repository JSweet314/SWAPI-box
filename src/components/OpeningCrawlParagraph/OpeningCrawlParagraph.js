import React from 'react';
import './OpeningCrawlParagraph.css'
import PropTypes from 'prop-types';

const OpeningCrawlParagraph = ({paragraph}) => {
  const lines = paragraph.split('\r\n').map((line, index) => {
    return <span className="crawl-line" key={index}>{line}</span>
  })
  return (
    <p>
      {lines}
    </p>
  );
};

OpeningCrawlParagraph.propTypes = {
  paragraph: PropTypes.string
}

export default OpeningCrawlParagraph;