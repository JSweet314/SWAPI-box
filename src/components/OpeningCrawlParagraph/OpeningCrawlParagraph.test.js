import React from 'react';
import OpeningCrawlParagraph from './OpeningCrawlParagraph';
import { shallow } from 'enzyme';

describe('OpeningCrawlParagraph', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(<OpeningCrawlParagraph paragraph={''}/>);
    expect(wrapper).toMatchSnapshot();
  });
});