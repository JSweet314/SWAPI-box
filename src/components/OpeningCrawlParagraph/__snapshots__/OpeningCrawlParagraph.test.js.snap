// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningCrawlParagraph should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningCrawlParagraph
    paragraph=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="crawl-line"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "crawl-line",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="crawl-line"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "crawl-line",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
